pkg/controller/servicebinding/secret.go

```diff
@@ -96,12 +160,15 @@ func newSecret(
 	client dynamic.Interface,
 	ns string,
 	name string,
+	payload map[string][]byte,
```

Comment:
The payload wasn't added here initially because the payload wasn't completely known at the time `newSecret` was called; has this changed?

Reply:
We did save the `envVars` in `options.binding.envVars` here -> https://github.com/redhat-developer/service-binding-operator/pull/750/files#diff-17972c83fa5f6d6588fce52545087fd259379261db8717e2057a37c1004f7d86L452

I moved it to newSecret(...) here -> https://github.com/redhat-developer/service-binding-operator/pull/750/files#diff-17972c83fa5f6d6588fce52545087fd259379261db8717e2057a37c1004f7d86R469

Reply:
You stated what you've done, not why you've done it ;)

Reply:
Eventually, later in the code, we put `b.envVars` as the secret data -> https://github.com/redhat-developer/service-binding-operator/pull/750/files#diff-17972c83fa5f6d6588fce52545087fd259379261db8717e2057a37c1004f7d86L301 just as it is.. no change to b.envVars.

We are anyway creating a new secret -> https://github.com/redhat-developer/service-binding-operator/pull/750/files#diff-17972c83fa5f6d6588fce52545087fd259379261db8717e2057a37c1004f7d86R465 and assigning SBR name to the binding secret. 

It's not the final name, but we do assign it early.

So, it may not be the final payload, but why not assign it at that point itself i.e. when we are calling newSecret(...) because eventually those envVars go into the binding secret, right?

Am I missing some use-case :) @isutton ?

Reply:
And it just seemed more logical to compare `b.secret.data` with SBR secret rather than `b.envVars`

Reply:
> So, it may not be the final payload, but why not assign it at that point itself i.e. when we are calling newSecret(...) because eventually those envVars go into the binding secret, right?

I'd argue that this secret handling could be simplified even further, perhaps with `newSecret` being called when the update is performed, since all its parameters are either available, or can easily be made available.

Reply:
Can we handle the secret handling in another PR?

Reply:
Being you the author of this PR, I'll leave that judgement to you :)

Reply:
Okay. Will handle that in another PR. Created an issue here https://github.com/redhat-developer/service-binding-operator/issues/822